### ISSUE TITLE
Invalid results when fast_inequality_join=true and table contains null values

### ISSUE THREAD

**Issue by sopel39** [prestodb/presto#8025]
```
presto:default> create table t3(a bigint, b bigint);
CREATE TABLE

presto:default> insert into t3 values (0,0), (0, -1), (0, null), (0, 1);
INSERT: 4 rows

presto:default> set session fast_inequality_join=true;
SET SESSION

presto:default> select * from nation n, t3 where n.regionkey -100 < t3.b and n.nationkey = t3.a;
 nationkey |   name    | regionkey |                                   comment                                    | a | b  
-----------+-----------+-----------+------------------------------------------------------------------------------+---+----
         0 | ALGERIA   |         0 |  haggle. carefully final deposits detect slyly agai                          | 0 | -1 
         0 | ALGERIA   |         0 |  haggle. carefully final deposits detect slyly agai                          | 0 |  0 
         1 | ARGENTINA |         1 | al foxes promise slyly according to the regular accounts. bold requests alon | 1 |  1 
         1 | ARGENTINA |         1 | al foxes promise slyly according to the regular accounts. bold requests alon | 1 |  2 
         1 | ARGENTINA |         1 | al foxes promise slyly according to the regular accounts. bold requests alon | 1 |  3 

presto:default> set session fast_inequality_join=false;
SET SESSION

presto:default> select * from nation n, t3 where n.regionkey -100 < t3.b and n.nationkey = t3.a;
 nationkey |   name    | regionkey |                                   comment                                    | a | b  
-----------+-----------+-----------+------------------------------------------------------------------------------+---+----
         0 | ALGERIA   |         0 |  haggle. carefully final deposits detect slyly agai                          | 0 |  1 
         0 | ALGERIA   |         0 |  haggle. carefully final deposits detect slyly agai                          | 0 | -1 
         0 | ALGERIA   |         0 |  haggle. carefully final deposits detect slyly agai                          | 0 |  0 
         1 | ARGENTINA |         1 | al foxes promise slyly according to the regular accounts. bold requests alon | 1 |  3 
         1 | ARGENTINA |         1 | al foxes promise slyly according to the regular accounts. bold requests alon | 1 |  2 
         1 | ARGENTINA |         1 | al foxes promise slyly according to the regular accounts. bold requests alon | 1 |  1 
```

Rows with null values are part of `SortedPositionLinks`. However, `applyLessThanFunction` will return `false` for null rows prematurely breaking join position enumeration.

This most likely is a release blocker since `fast_inequality_join` is true by default.

FYI: @pnowojski @dain @martint

**Comment by sopel39** [comment]
The proposed fix is to not add null rows to `SortedPositionLinks`

**Comment by electrum** [comment]
Looks like this was introduced in 0.174

**Comment by electrum** [comment]
The session property seems to be false by default. Am I missing something? https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java#L244

**Comment by pnowojski** [comment]
@electrum unfortunately it is enabled by default. This `false` that you pointed is for `hidden` attribute.

**Comment by sopel39** [comment]
Fix: https://github.com/prestodb/presto/pull/8036

**Comment by raghavsethi** [comment]
@pnojowski: Why is an experimental feature enabled by default?

**Comment by pnowojski** [comment]
In initial PR that was disabled by default and only later during reviewing process it was proposed to be enable this by default (https://github.com/prestodb/presto/pull/7097#discussion_r111239527). I guess at that point I forgot to change this from an experimental feature :(

**Comment by dain** [comment]
Fixed by #8036

**Comment by findepi** [comment]
@raghavsethi @pnowojski  since this is not meant to be flagged as experimental feature, let's remove "experimental" from the flag description (#8079)
